doc/api_plugins.rst

```diff
@@ -43,27 +43,53 @@ When DNF CLI runs it loads the plugins found in the paths during the CLI's initi
 
     Plugin must override this. Called immediately after all the plugins are loaded. `base` is an instance of :class:`dnf.Base`. `cli` is an instance of :class:`dnf.cli.Cli` but can also be ``None`` in case DNF is running without a CLI (e.g. from an extension).
 
-  .. method:: config
+  .. method:: pre_config()
+
+    Plugin can override this. This hook is called before configuring the repos.
+
+  .. method:: config()
 
     Plugin can override this. This hook is called immediately after the CLI/extension is finished configuring DNF.  The plugin can use this to tweak the global configuration or the repository configuration.
```

Comment:
Quite unrelated, but if you wanted to improve this a bit further I'd remove all the occurrences of _Plugin can override this._ as I think it's clear from the context.

For the few other descriptions that start with _Plugin does ..._, it should read _The plugin does ..._.